### PR TITLE
WIP: Update contrasts to new pybids format

### PR DESCRIPTION
**Pull request by adelavega**
This is WIP but I want to start a discussion on how to go forward. So far I only have some minor syntactic changes.

Previously, pybids returned a single contrast matrix, with each row representing an input from the previous level. E.g. if you had 3 runs for one subject, and one contrasts; at the session level you would have 3 rows. Usually these would be all 1s, as you are trying to take the average of the 3 runs at the session level.
E.g. 
```
contrasts:
   RT-trial_type
0              1
1              1
2              1
```
```
index:
   run subject              task    
0    1      01  mixedgamblestask 
1    2      01  mixedgamblestask 
2    3      01  mixedgamblestask
```

Pybids now returns NamedTuples in order to keep the weights for each contrast separate, in order to accomodate f-tests (which would not fit on this table).

The other difference, is that now the weights are not repeated across input units. In the previous example, you would only have 1 "row". 
e.g.
```
In [34]: contrasts[0].weights
Out[34]: 
   RT-trial_type
0              1
```
This contrast applies to all inputs with the following entities
```
entities={'subject': '01', 'task': 'mixedgamblestask'})
```

So basically, I'm trying to figure out if we should emulate the way contrasts were previously represented by expanding out the single row to three... or if there is a more efficient way to do this that makes sense.

Relatedly, we should probably also tack F-tests here, which probably means we shouldn't pass around a single matrix between nipype interfaces (as we wont be able to w/ t- and F tests mixed together.

I think the only difference here is that F tests have multiple rows. I also assume these will mostly be used at the first-level (although not exclusively).
E.g an F-test testing for any effect in RT or Accuracy
```
RT Accuracy Condition
1       0           0
0       1           0
```

### CODE REVIEW

File: fitlins/workflows/base.py [hunk 1]
```diff
@@ -269,11 +277,11 @@ def _get_evs(info):
         (l1_model, plot_corr, [('design_matrix', 'data')]),
         (get_evs, plot_corr, [('out', 'explanatory_variables')]),
```
Comment (effigies) [inline fitlins/workflows/base.py]: I'm going to push to your branch with a commit blocking all plotting for now. `get_evs` is going to take rethinking, as we'll need to derive the explanatory variables from the contrasts.

Just a heads up.

Comment (adelavega) [reply fitlins/workflows/base.py]: Yep I was thinking the same thing

**Comment by effigies** [comment]
Just a heads up I pushed a few fixes.

**Comment by adelavega** [comment]
hey @effigies I got the first level running. I think in theory this should work for F tests (they are just nxn instead of 1xn matrices). The nice thing about this way of passing around contrasts is that it looks more like how nipype does it w/ FSL (e.g. condition names, and weights for those). We could change the specific format to be more "neutral" if you wanted). 

Please review!

Next up is taking a look at higher levels. It will require "expanding" contrasts, since only one is given for a group of images (e.g. all runs in subject) rather than one per input. Conceptually, if I fix that it should work.

**Comment by effigies** [comment]
@adelavega Thanks. I'll have a look ASAP.
